common/ledger/blockledger/fileledger/factory_test.go

```diff
@@ -7,74 +7,83 @@ SPDX-License-Identifier: Apache-2.0
+			t,
+			"boogie",
+			func() { f.ChannelIDs() },
+			"Expected ChannelIDs to panic if storage provider cannot list channel IDs",
+		)
+	})
 
-func TestBlockstoreProviderError(t *testing.T) {
-	flf := &fileLedgerFactory{
-		blkstorageProvider: &mockBlockStoreProvider{error: fmt.Errorf("blockstorage provider error")},
-		ledgers:            make(map[string]blockledger.ReadWriter),
-	}
-	require.Panics(
-		t,
-		func() { flf.ChannelIDs() },
-		"Expected ChannelIDs to panic if storage provider cannot list channel IDs")
+	t.Run("open", func(t *testing.T) {
+		mockBlockStore.OpenReturns(nil, errors.New("woogie"))
+		_, err := f.GetOrCreate("foo")
+		require.EqualError(t, err, "woogie")
+		require.Empty(t, f.ledgers, "Expected no new ledger is created")
+	})
 
-	_, err := flf.GetOrCreate("foo")
-	require.Error(t, err, "Expected GetOrCreate to return error if blockstorage provider cannot open")
-	require.Empty(t, flf.ledgers, "Expected no new ledger is created")
+	t.Run("remove", func(t *testing.T) {
+		mockBlockStore.DropReturns(errors.New("oogie"))
+		err := f.Remove("foo")
+		require.EqualError(t, err, "oogie")
+	})
 }
 
 func TestMultiReinitialization(t *testing.T) {
 	metricsProvider := &disabled.Provider{}
 
-	dir, err := ioutil.TempDir("", "hyperledger_fabric")
+	dir, err := ioutil.TempDir("", "fileledger")
 	require.NoError(t, err, "Error creating temp dir: %s", err)
 	defer os.RemoveAll(dir)
 
-	flf, err := New(dir, metricsProvider)
+	f, err := New(dir, metricsProvider)
 	require.NoError(t, err)
-	_, err = flf.GetOrCreate("testchannelid")
+	_, err = f.GetOrCreate("testchannelid")
 	require.NoError(t, err, "Error GetOrCreate channel")
-	require.Equal(t, 1, len(flf.ChannelIDs()), "Expected 1 channel")
-	flf.Close()
+	require.Equal(t, 1, len(f.ChannelIDs()), "Expected 1 channel")
+	f.Close()
 
-	flf, err = New(dir, metricsProvider)
+	f, err = New(dir, metricsProvider)
 	require.NoError(t, err)
-	_, err = flf.GetOrCreate("foo")
+	_, err = f.GetOrCreate("foo")
 	require.NoError(t, err, "Error creating channel")
-	require.Equal(t, 2, len(flf.ChannelIDs()), "Expected channel to be recovered")
-	flf.Close()
+	require.Equal(t, 2, len(f.ChannelIDs()), "Expected channel to be recovered")
+	f.Close()
 
-	flf, err = New(dir, metricsProvider)
+	f, err = New(dir, metricsProvider)
 	require.NoError(t, err)
-	_, err = flf.GetOrCreate("bar")
+	_, err = f.GetOrCreate("bar")
 	require.NoError(t, err, "Error creating channel")
-	require.Equal(t, 3, len(flf.ChannelIDs()), "Expected channel to be recovered")
-	flf.Close()
+	require.Equal(t, 3, len(f.ChannelIDs()), "Expected channel to be recovered")
+	f.Close()
+
+	f, err = New(dir, metricsProvider)
+	require.NoError(t, err)
+	err = f.Remove("bar")
+	require.NoError(t, err, "Error removing channel")
+	require.Equal(t, 2, len(f.ChannelIDs()))
+	err = f.Remove("this-isnt-an-existing-channel")
+	require.NoError(t, err, "Error removing channel")
+	require.Equal(t, 2, len(f.ChannelIDs()))
+	f.Close()
```

Comment:
I am not sure whether this test fits in `TestMultiReinitialization()`. Maybe `TestRemove()`. Please fix this in the subsequent PR.